plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -250,6 +251,29 @@ public void setColumnComment(SchemaTableName schemaTableName, String columnName,
         tableCache.invalidate(schemaTableName);
     }
 
+    public void renameTable(SchemaTableName oldName, SchemaTableName newName)
+    {
+        String oldSchemaName = toRemoteSchemaName(oldName.getSchemaName());
+        String oldTableName = toRemoteTableName(oldSchemaName, oldName.getTableName());
+        String newSchemaName = toRemoteSchemaName(newName.getSchemaName());
+
+        // Need to check explicitly because the old collection may not exist when it doesn't have any data
+        if (collectionExists(client.getDatabase(oldSchemaName), oldTableName)) {
+            getCollection(oldName).renameCollection(new MongoNamespace(newSchemaName, newName.getTableName()));
+        }
+
+        // Schema collection should always have the source table definition
+        MongoCollection<Document> oldSchema = client.getDatabase(oldSchemaName).getCollection(schemaCollection);
```

Comment:
What transactional consistency guarantees does the mongodb connector currently give? 
EG) A trino node crashing in between delete/insert here

mongo has multi document transactions in v4.0 and greater

Reply:
The current isolation level is a single document. It's basically pre-exiting issue in the connector. Let me revisit after upgrading the minimum version.